Main.cpp

```diff
@@ -191,43 +239,63 @@ int main() {
       // canBus.printRxAll();
     }
 
-    chThdSleepMilliseconds(50);
+    chThdSleepMilliseconds(10000);
   }
 }
 
-// module utility functions
 // @brief Convert bytes in RX buffer from ADC reading to temperature
 // @return temp Temperature in degrees C, resolution of .25
+// TODO: Add in per-channel moving average
 uint8_t adc_to_temp(uint8_t* rxbuf) {
   static uint16_t raw{};
-  static int norm{};
+  static double voltage{};
```

Comment:
Well, the code would be easier to follow if you declared variables as you used them instead of declaring them all at the top of the function. gcc.godbolt.org seems to show it optimizing it all away anyway.